test/src/postprocessors/DisplacedMeshBlockVolume.C

```diff
@@ -0,0 +1,54 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "DisplacedMeshBlockVolume.h"
+
+#include "DisplacedProblem.h"
+#include "MooseMesh.h"
+
+registerMooseObject("MooseTestApp", DisplacedMeshBlockVolume);
+
+InputParameters
+DisplacedMeshBlockVolume::validParams()
+{
+  InputParameters params = GeneralPostprocessor::validParams();
+  params.addParam<std::vector<SubdomainName>>("subdomains",
+                                              "Subdomains to be considered (default means all)");
+  return params;
+}
+
+DisplacedMeshBlockVolume::DisplacedMeshBlockVolume(const InputParameters & parameters)
+  : GeneralPostprocessor(parameters), _mesh(_fe_problem.getDisplacedProblem()->mesh())
+{
+  if (isParamValid("subdomains"))
+  {
+    auto blocks = _mesh.getSubdomainIDs(getParam<std::vector<SubdomainName>>("subdomains"));
+    for (const auto & blk : blocks)
+      _blocks.insert(blk);
+  }
+  else
+    _blocks = _mesh.meshSubdomains();
```

Comment:
why are you offering two options for a test object?

Reply:
just delete the option we dont need to re-implement block restriction for a generalUO that would normally be an elementUO